src/chain/tezos/contracts/DexterPoolHelper.ts

```diff
@@ -103,6 +103,34 @@ export namespace DexterPoolHelper {
         }
     }
 
+    /**
+     * 
+     * @param server Destination Tezos node.
+     * @param address Pool contract address.
+     * @param account 
+     * @returns Raw values that are not scaled.
+     */
+    export async function getAccountPoolShare(server: string, address: string, account: string): Promise<{token: number, xtz: number}> {
+        try {
+            const storage = await getSimpleStorage(server, address);
+
+            const packedKey = TezosMessageUtils.encodeBigMapKey(Buffer.from(TezosMessageUtils.writePackedData(account, 'address'), 'hex'));
+            const mapResult = await TezosNodeReader.getValueForBigMapKey(server, storage.balanceMap, packedKey);
+
+            if (mapResult === undefined) { throw new Error(`Map ${storage.balanceMap} does not contain a record for ${account}`); }
```

Comment:
We are throwing errors to short circuit a try/catch block but the error is not consumed in any way. I assume the motivation here was to have fixed return paths but this looks odd. Perhaps we can log the error in the `catch` to make it somewhat meaningful?

Reply:
This code is presently not in use.

Reply:
Regardless, this is not a good pattern. We should avoid it if possible.